UniversityDatabase/Database.cpp

```diff
@@ -16,61 +68,125 @@ void Database::RemovePersonByPesel(const std::string pesel)
 	{
 		if (pesel == it->GetPesel())
 		{
-			RemovePerson(it);
+			Persons.erase(it);
 			break;
 		}
 	}
 }
 
-void Database::RemovePerson(People::iterator it)
-{
-	Persons.erase(it);
-}
-
 People Database::SearchBySurname(const std::string surname)
 {
 	People foundPeople;
-	for (People::iterator it = Persons.begin(); it < Persons.end(); it++)
+	for (auto person : Persons)
 	{
-		if (surname == it->GetSurname())
+		if (surname == person.GetSurname())
 		{
-			foundPeople.push_back(*it);
+			foundPeople.push_back(person);
 		}
 	}
 	return foundPeople;
 }
 
-Person Database::SearchByPesel(const std::string pesel)
+Person* Database::SearchByPesel(const std::string pesel)
 {
-	for (People::iterator it = Persons.begin(); it < Persons.end(); it++)
+	for (auto person : Persons)
 	{
-		if (pesel == it->GetPesel())
+		if (pesel == person.GetPesel())
 		{
-			return *it;
+			Person savedPerson = person;
+			return &savedPerson;
 		}
 	}
+	return NULL;
+}
+
+void Database::ModifyAddressByPesel(const std::string pesel, const std::string newAddress)
+{
+	Person* foundPerson = SearchByPesel(pesel);
+	if (foundPerson != NULL)
+	{
+		Person personToChange = *foundPerson;
+		personToChange.ChangeAddress(newAddress);
+		std::replace(Persons.begin(), Persons.end(), *foundPerson, personToChange);
```

Comment:
https://www.techiedelight.com/compare-two-objects-in-cpp/

Add below in Person.h:
```
class Person
{
...
	Person(std::string pesel) : pesel(pesel) {} //nalezy przeładować operator aby kod się kompilował
	bool operator == (const Person& rhs) const;
};

bool Person::operator == (const Person& rhs) const {
	return pesel == rhs.pesel;
}
```